### ISSUE TITLE
Leave an instance

### ISSUE THREAD

**Issue by teagan75** [OnapleRPG/EpicBoundaries#2]
How does one leave an instance to go back to main world? Other than exiting game and reconnecting?

**Comment by obr-09** [comment]
The `apparate` command should teleport you to the specified world. With the default overworld name, `/apparate world` or `/apparate` can bring you back to the overworld.  
The full command is `/apparate <world-name> <x y z> <player>`, with every parameter being optional.  
(By default, overworld is named "world", the nether "DIM-1" and the end "DIM1" - once generated)  

The readme could mention those informations about default world names though; I will update it once I can verify it

**Comment by teagan75** [comment]
ok it isn't working on single player. will have to check server which is where I will be using this function anyways

**Comment by obr-09** [comment]
By single player, do you mean a local server? Because it cannot work simply on a client.  
If so do not hesitate to provide logs if you have some errors in server console ; or even from where you input the command

**Comment by teagan75** [comment]
It actually does work on a client :) other than leaving the instance. I tried on a dedicated server and this is what shows in the console when running the command /apparate world from the secondary world

31.12 10:15:30 [Server] Server thread/ERROR [Sponge]: Unable to load world [./world/DIM2/world]. We cannot find its folder under [./world/DIM2].

Also, I need to specify all the arguments in the command or it doesn't work which is fine :) world still doesn't work because it apparently is looking for the default world in the dimension folder.

**Comment by teagan75** [comment]
So, what I am thinking is the issue is that the default world cannot be a world players return to? It cannot be my lobby. I need to make another dimension folder as my lobby

Edit: nvm it is all of the sudden working 🤦

**Comment by obr-09** [comment]
That's interesting, we built it for a server plugin, but didn't think it would work on a client.  
There are calls to sponge api "getServer" method, so I don't really know how a client behaves in that case...   

However, from what you're saying, it simply does not find the world, like the file structure is different between client and server... 
And that's even stranger that it suddenly works  😕 I will try to see how it behaves on a client  
_edit: wasn't it just that your world has a custom name different than "world"? It takes the name you create your game with. If so, then the appropriate command would be `/apparate YourWorldName x y z`_

**Comment by teagan75** [comment]
/apparate world isn't working again on the server. My world is named world in the level,dat. The console says it is looking for the world in the dimension folder

03.01 15:32:31 [Server] Server thread/ERROR [Sponge]: Unable to load world [./world/DIM3/world]. We cannot find its folder under [./world/DIM3].

**Comment by teagan75** [comment]
So, I keep having to alter the level.dat for each world I want because it doesn't like the name it comes with. Then the command works

**Comment by obr-09** [comment]
So this time it is on a real server?  
There seems to be a problem with relative path, it's looking from within the DIM3 folder whereas it should not.  
`it doesn't like the name it comes with` May I ask you to provide an example of what name it sets, and what do you change it to?

**Comment by teagan75** [comment]
Yes on a real server this time. So, the level.dat had the name set as saves and I changed it to monster and then I could use the /apparate world command. I really don't know why that would make a difference but it does. When I said it all of the sudden worked, I didn't realize that is what made it work, since I was trying out so many other things. Then I added another world and had the same issue of the command not working. I changed the level.dat name again and it worked again. 

I am testing my worlds out on single player before moving them to the server. They were named different things but I always need to change it. So, it is probably because I am testing on sp idk but I figured it out. Thanks for all your help

**Comment by obr-09** [comment]
I did a few tests; glad you were able to solve the issue  
Copying a SP save file already named is fine, but manually renaming world folders require level.dat editing.  
I added some paragraphs [to the readme](https://github.com/OnapleRPG/EpicBoundaries#preparing-worlds). I also found this convenient [online NBT editor](https://irath96.github.io/webNBT/) which could prove useful.  

Closing this, thanks for your feedback
